### PR TITLE
fix bug: updating the password for a non-existing user does not report error

### PR DESCRIPTION
**Pull request by jiangke-sr**
Signed-off-by: kejiang <ke.jiang@zilliz.com>

issue: #16504

1. FIX: updating password for a non-existing user does not report error
2. ADD: updating password will check the old password
3. ADD: toggle for enabling user authentication (default not enabled), replace the logic for if there are created users

### CODE REVIEW

**Comment by aakejiang** [comment]
rerun ut

**Comment by aakejiang** [comment]
rerun ut

File: internal/rootcoord/meta_table.go [hunk 1]
```diff
@@ -1360,7 +1360,12 @@ func (mt *MetaTable) AddCredential(credInfo *internalpb.CredentialInfo) error {
 		return fmt.Errorf("username is empty")
 	}
 	k := fmt.Sprintf("%s/%s", CredentialPrefix, credInfo.Username)
-	err := mt.txn.Save(k, credInfo.EncryptedPassword)
+	v, err := proto.Marshal(&internalpb.CredentialInfo{EncryptedPassword: credInfo.EncryptedPassword})
+	if err != nil {
+		log.Error("MetaTable marshal credential info fail", zap.String("key", k), zap.Error(err))
```
Comment (jaime0815) [inline internal/rootcoord/meta_table.go:1365]: is it should be WARN level?

**Comment by xiaofan-luan** [comment]
> Signed-off-by: kejiang [ke.jiang@zilliz.com](mailto:ke.jiang@zilliz.com)

You need to specify the issue you are trying to fix. Better to have some details about the fix

**Pull request merged**